### DISCUSSION TITLE
Remote read .pgrb2 files

### DISCUSSION THREAD

**Discussion by valpo-syncretis** [Unidata/netcdf-java#1058] [Questions and Answers] [answered]
Hello!

I'm trying to remotely read .pgrb2 file and get this:
java.io.IOException: java.io.IOException: Cant read https://ftp.ncep.noaa.gov/data/nccf/com/gfs/prod/gfs.20220809/00/atmos/gfs.t00z.pgrb2.0p25.f000: not a valid CDM file.

NetcdfFile file = NetcdfFiles.open("https://ftp.ncep.noaa.gov/data/nccf/com/gfs/prod/gfs.20220809/00/atmos/gfs.t00z.pgrb2.0p25.f000");

I have only grib latest artifact and for local read it works. Should I add something else to pom? Sorry for so silly question.

**Comment by lesserwhirls** [comment, marked as answer]
Unfortunately, netCDF-Java does not currently support reading remote GRIB files via http (or AWS S3) at this point. Code modifications are needed to handle the creation and use of GRIB index files for remote GRIB files. See https://github.com/Unidata/netcdf-java/issues/638 for some of the discussion on the topic.

**Reply by valpo-syncretis** [reply]
Thank you very much for the answer. Anyway I'm looking forward to hearing from develop team.

**Comment by JohnLCaron** [comment]
you could ask ncep to run a thredds server on that data, or check if they already do.

**Comment by JohnLCaron** [comment]
the obvious thing to do is to ask ncep to run a thredds server on that
data, or check if they already do.
